core/sbt-cloudflow/src/main/scala/cloudflow/sbt/BuildAppPlugin.scala

```diff
@@ -93,7 +93,8 @@ object BuildAppPlugin extends AutoPlugin {
 
     // need to get the proper image name in `StreamletDeployment` s too
     val newDeployments = appDescriptor.deployments.map { deployment =>
-      deployment.copy(image = streamletClassNamesToImageNameAndId.get(deployment.className).get.imageName.toString)
+      val image = streamletClassNamesToImageNameAndId.get(deployment.className).get
+      deployment.copy(image = s"${image.imageName.toString().split(":").head}@${image.imageId.toString()}")
```

Comment:
Prefer `imageName.copy(tag = Some(imageAndId.imageDigest.toString())).toString()`

Reply:
Also please rename `ImageNameAndId` to `ImageNameAndDigest`

Reply:
Right, so we cannot do that since the built-in ´toString´ for the ImageName class just knows about tags and not digests.

Reply:
Ok, I thought it was `image:@sha256:<sha>`, but seems it is `image@sha256:<sha>`. Still it is better to add this logic to ImageName, even as an extension method if needed, and reuse the logic that is there, the imageName knows about the parts.

Reply:
would be nice if you could do `imageName.referenceWithDigest(imageDigest)` or something like that. Add the method to an implicit class so you are able to use the fields of ImageName.

Reply:
Done, put it in its own file.